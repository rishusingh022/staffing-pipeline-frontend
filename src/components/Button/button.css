.button {
  background-color: #2251ff;
  border: none;
  color:#ffffff;
  padding: 11.5px 12px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
}
