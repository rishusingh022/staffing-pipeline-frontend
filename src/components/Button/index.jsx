import React from 'react';
import './button.css';
import PropTypes from 'prop-types';

export default function Button({ buttonText, handleClick }) {
  return (
    <button onClick={handleClick} className="button" data-testid="button">
      {buttonText}
    </button>
  );
}

Button.propTypes = {
  buttonText: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};
