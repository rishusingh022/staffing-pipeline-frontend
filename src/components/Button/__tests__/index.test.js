import React from 'react';
import Button from '..';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Button Component', () => {
  it('should render correctly', () => {
    const handleClick = jest.fn();
    const { asFragment } = render(<Button buttonText="search" handleClick={handleClick} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call the handleClick functionality when clicked', () => {
    const handleClick = jest.fn();
    render(<Button buttonText="search" handleClick={handleClick} />);
    const button = screen.getByTestId('button');
    fireEvent.click(button);
    expect(handleClick).toBeCalled();
  });
});
