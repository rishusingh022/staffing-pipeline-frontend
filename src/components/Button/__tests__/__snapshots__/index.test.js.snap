// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<DocumentFragment>
  <button
    class="button"
    data-testid="button"
  >
    search
  </button>
</DocumentFragment>
`;
